src/session/pty_session.rs

```diff
@@ -0,0 +1,254 @@
+#[cfg(not(feature = "async"))]
+mod sync {
+    use std::io::{self, BufRead, Read, Stdout, Write};
+    use crate::{
+        process::unix::{PtyStream, UnixProcess},
+        session::{Session, OsProcess},
+        stream::{log::LogStream, StreamSink},
+        Captures, Needle,
+    };
+    
+    // TODO(tmpfs): fix this type for windows
+    type LogSession = Session<UnixProcess, LogStream<PtyStream, Stdout>>;
+
+    /// Wraps a session that may be logged to stdout.
+    #[derive(Debug)]
+    pub enum PtySession {
+        /// Default pty session.
+        Default(Session),
+        /// Pty session that logs to stdout.
+        Logged(LogSession),
+    }
```

Comment:
I think it's implementaion details that we have an internal enum; I'd rather hide it as an inner enum of pub struct, and create a 2 creation methods?

Reply:
Sure we could change `ReplSession::new_pty` to `ReplSession::new_log` and make `PtySession` a private type, how does that sound?

Reply:
I have pushed a commit that improves this, so `PtySession` is not involved in the constructors, we can't make it `pub(crate)` because we need it to be an associated type for a `Deref` implementation ([here](https://github.com/tmpfs/expectrl/blob/pty-session/src/repl.rs#L353)) so instead I have opted for `#[doc(hidden)]` as it is now an implementation detail.